src/main/java/frc/robot/systems/ClimberMechFSM.java

```diff
@@ -0,0 +1,248 @@
+package frc.robot.systems;
+
+// WPILib Imports
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+// Third party Hardware Imports
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkLimitSwitch;
+import com.revrobotics.SparkLimitSwitch.Type;
+
+// Robot Imports
+import frc.robot.TeleopInput;
+import frc.robot.HardwareMap;
+
+public class ClimberMechFSM {
+	/* ======================== Constants ======================== */
+
+	// FSM state definitions
+	public enum ClimberMechFSMState {
+		START_STATE,
+		RAISE_HOOK_MANUAL,
+		LOWER_HOOK_MANUAL,
+		IDLE
+	}
+
+	// The ranges for different powers for the step function as a set of x, y pairs
```

Comment:
x and y are bad variable names here, be more specific on whatinput and output this is using.